flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamOverAggregate.scala

```diff
@@ -119,6 +154,66 @@ class DataStreamOverAggregate(
 
   }
 
+  def createTimeBoundedProcessingTimeOverWindow(
+    inputDS: DataStream[Row]): DataStream[Row] = {
+
+    val overWindow: Group = logicWindow.groups.get(0)
+    val partitionKeys: Array[Int] = overWindow.keys.toArray
+    val namedAggregates: Seq[CalcitePair[AggregateCall, String]] = generateNamedAggregates
+
+    val index = overWindow.lowerBound.getOffset.asInstanceOf[RexInputRef].getIndex
+    val count = input.getRowType().getFieldCount()
+    val lowerboundIndex = index - count
+    var time_boundary = 0L
+     
+    logicWindow.constants.get(lowerboundIndex).getValue2 match {
```

Comment:
This can be changed to
```
val timeBoundary = logicWindow.constants.get(lowerboundIndex).getValue2 match {
  case _: java.math.BigDecimal => logicWindow.constants.get(lowerboundIndex)
    .getValue2.asInstanceOf[java.math.BigDecimal].longValue()
  case _ => throw new TableException("OVER Window boundaries must be numeric")
}
```